Sources/SignHereLibrary/Commands/CreateProvisioningProfileCommand.swift

```diff
@@ -286,18 +295,39 @@ internal struct CreateProvisioningProfileCommand: ParsableCommand {
             certificateSigningRequestSubject: try container.decode(String.self, forKey: .certificateSigningRequestSubject),
             bundleIdentifierName: try container.decodeIfPresent(String.self, forKey: .bundleIdentifierName),
             platform: try container.decode(String.self, forKey: .platform),
-            profileName: try container.decodeIfPresent(String.self, forKey: .profileName)
+            profileName: try container.decodeIfPresent(String.self, forKey: .profileName),
+            autoRegenerate: try container.decode(Bool.self, forKey: .autoRegenerate)
         )
     }
 
     internal func run() throws {
-        let privateKey: Path = .init(privateKeyPath)
-        let csr: Path = try createCSR(privateKey: privateKey)
         let jsonWebToken: String = try jsonWebTokenService.createToken(
             keyIdentifier: keyIdentifier,
             issuerID: issuerID,
             secretKey: try files.read(Path(itunesConnectKeyPath))
         )
+        let deviceIDs: Set<String> = try iTunesConnectService.fetchITCDeviceIDs(jsonWebToken: jsonWebToken)
+        if autoRegenerate, profileName == nil {
```

Comment:
Should this just be performed in a validate function? https://apple.github.io/swift-argument-parser/documentation/argumentparser/validation/

Reply:
sure! done